phys/module_diag_functions.F

```diff
@@ -436,8 +439,8 @@ END FUNCTION Pwat
   !~    to higher levels.                                                     ~!
   !~                                                                          ~!
   !!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
-    FUNCTION Buoyancy ( nz, tk, rh, p, hgt, sfc, cape, cin, zlfc, plfc, lidx,  &
-                        parcel ) result (ostat)
+    FUNCTION Buoyancy ( nz, tk, rh, p, hgt, sfc, cape, cin, zlcl, zlfc, zel, zmu,  &
+                        parcel ) result (ostat) !Add declaration of lcl, el, and MU layer. Modified by Zhixiao.
```

Comment:
I see the arguments changed but not the function call

Reply:
@davegill @dudhia Yeah, phys/module_diag_afwa.F and Registry/registry.afwa are recommended to be consistently modified based on this argument changes in buoyancy function. Despite this, we conducted other thousands of lines changes that are specialized for our simulation, but can not be generalized for public usages. To less confuse the public, I try to avoid to do the comprehensive framing modifications and only uploaded the module_diag_functions.F.

Reply:
@davegill @dudhia @zhixiaozhang To clarify, I believe Zhixiao uploaded all of his code changes to module_diag_functions.F but these require changes to the driver and registry files because of new variables (height of most unstable layer (zmu), equilibrium level (zel), and LCL (zlcl). Those new variables are not a bug fix though and are an add on that may or may not be useful. I don't know if those go through a different process to implement added diagnostics. If the bug fixes alone are to be implemented (the most unstable (MU) layer definition and the calculation of precipitable water), then all of the other code related to new variables needs to be removed, and then the file should work. if that is what is wanted, I believe Zhixiao can do that without too much work. If instead, the added variables are also wanted, then Zhixiao can supply the other modified driver and registry files.